src/components/studies/app-design/AppDesign.tsx

```diff
@@ -316,6 +332,88 @@ const AppDesign: React.FunctionComponent<
     isIrbEmailValid: true,
   })
 
+  const [errorState, setErrorState] = React.useState<ErrorStateType>({})
+
+  const handleUpdate = (updatedStudy: Study) => {
+    const formattedStudy = formatStudy(updatedStudy)
+    if (formattedStudy.contacts) {
+      for (const contact of formattedStudy.contacts) {
+        if (contact.name === '') {
+          contact.name = DEFAULT_CONTACT_NAME
+        }
+      }
+    }
+    onUpdate(formattedStudy)
+  }
+
+  const checkPhoneError = (contactLead?: Contact, irbRecord?: Contact) => {
+    const generalContactPhoneError =
+      !contactLead?.phone?.number || isInvalidPhone(generalContactPhoneNumber)
+    const irbRecordHasError =
+      !irbRecord?.phone?.number || isInvalidPhone(irbPhoneNumber)
+    setPhoneNumberErrorState({
+      isGeneralContactPhoneNumberValid: !generalContactPhoneError,
+      isIrbPhoneNumberValid: !irbRecordHasError,
+    })
+  }
+
+  const checkEmailError = (contactLead?: Contact, irbRecord?: Contact) => {
```

Comment:
You can make this more generic as well  
Would something like this work? I think you can do this with the phone as well
```
const checkEmailError = (contact: Contact, error: keyof ErrorStateType) {
    const isError =v ! contact.email  || isInvalidEmail(contact.email)
     setEmailErrorState(...emailErrorState, [error]: isError)
}
```

Reply:
Hey, so in order for this to work, I wold have to call the checkEmailError function twice, one for the contactLead and another for the irbRecord. I was doing this before and was getting weird behavior because we are setting the same state twice in a very close time frame to each other, so some things were getting overwritten. A option would just be to return isError from the method and update it together back in the useEffect.